uniswap-v2/src/lib.rs

```diff
@@ -192,16 +201,255 @@ fn map_swap_events(
+                );
+                continue;
+            }
+            Some(pair_created_event) => {
+                swaps.items.push(dex_amm::Swap {
+                    tx_hash: swap_event.tx_hash.clone(),
+                    log_index: swap_event.log_index,
+                    to: swap_event.to.clone(),
+                    from: swap_event.sender.clone(),
+                    block_number: block.number,
+                    timestamp: block.timestamp(),
+                    tokens_in: vec![
+                        pair_created_event.token0.clone(),
+                        pair_created_event.token1.clone(),
+                    ],
+                    amounts_in: vec![
+                        swap_event.amount0_in.clone(),
+                        swap_event.amount1_in.clone(),
+                    ],
+                    tokens_out: vec![
+                        pair_created_event.token0.clone(),
+                        pair_created_event.token1.clone(),
+                    ],
+                    amounts_out: vec![
+                        swap_event.amount0_out.clone(),
+                        swap_event.amount1_out.clone(),
+                    ],
+                    pool_address: pair_created_event.pair.clone(),
+                })
+            }
+        }
+    }
+
+    Ok(swaps)
+}
+
 #[substreams::handlers::store]
-fn store_swap_events(swap_events: uniswap::SwapEvents, output: store::StoreSetRaw) {
-    log::info!("Stored events {}", swap_events.items.len());
-    for event in swap_events.items {
+fn store_deposits(deposits: dex_amm::Deposits, output: store::StoreSetRaw) {
+    log::info!("Stored events {}", deposits.items.len());
+    for event in deposits.items {
+        let pool_key = keyer::deposit_key(&event.tx_hash, event.log_index);
+        output.set(0, &pool_key, &proto::encode(&event).unwrap());
+    }
+}
+
+#[substreams::handlers::store]
+fn store_withdraws(withdraws: dex_amm::Withdraws, output: store::StoreSetRaw) {
+    log::info!("Stored events {}", withdraws.items.len());
+    for event in withdraws.items {
+        let pool_key = keyer::withdraw_key(&event.tx_hash, event.log_index);
+        output.set(0, &pool_key, &proto::encode(&event).unwrap());
+    }
+}
+
+#[substreams::handlers::store]
+fn store_swaps(swaps: dex_amm::Swaps, output: store::StoreSetRaw) {
+    log::info!("Stored events {}", swaps.items.len());
+    for event in swaps.items {
         let pool_key = keyer::swap_key(&event.tx_hash, event.log_index);
         output.set(0, &pool_key, &proto::encode(&event).unwrap());
     }
 }
 
 #[substreams::handlers::store]
-fn store_swap_count(swap_events: uniswap::SwapEvents, s: store::StoreAddInt64) {
-    s.add(0, "total", swap_events.items.len() as i64);
+fn store_counts(clock: Clock, pair_created_events: uniswap::PairCreatedEvents, mint_events: uniswap::MintEvents, burn_events: uniswap::BurnEvents, swap_events: uniswap::SwapEvents, s: store::StoreAddInt64) {
```

Comment:
I wonder if there is a more generic way of implementing this (e.g. have a way to accumulate count for any repeated messages). @robinbernon do you have any thoughts on this?

Reply:
I just pushed an update that generalizes it a little more. Another option would be to track the counts as a field in the `Transaction` or `UsageEvents` proto message. 

More ideas welcome!